### PR TITLE
feat(jsx-renderer): Support "stream" option in jsxRenderer.

### PR DESCRIPTION
**Pull request by usualoma**
@yusukebe 
How about this?
Implemented: #1659

### Author should do the followings, if applicable

- [x] Add tests
- [x] Run tests
- [x] `yarn denoify` to generate files for Deno

### CODE REVIEW

File: src/middleware/jsx-renderer/index.ts [hunk 1]
```diff
@@ -23,16 +26,25 @@ const createRenderer =
         : options?.docType === true
         ? '<!DOCTYPE html>'
         : ''
+    const body = html`${raw(docType)}${jsx(
+      RequestContext.Provider,
+      { value: c },
+      (component ? component({ children, ...(props || {}) }) : children) as any
```
Comment (yusukebe) [inline src/middleware/jsx-renderer/index.ts:32]: Please add the flag:

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

File: src/middleware/jsx-renderer/index.ts [hunk 2]
```diff
@@ -23,16 +26,25 @@ const createRenderer =
...
-    )
+    if (options?.stream) {
+      return c.body(renderToReadableStream(body), {
+        headers:
+          options.stream === true
+            ? {
+                'Transfer-Encoding': 'chunked',
+                'Content-Type': 'text/html; charset=UTF-8',
+              }
+            : options.stream,
```
Comment (yusukebe) [inline src/middleware/jsx-renderer/index.ts:43]: This is good!

**Review by yusukebe — COMMENTED** [review]
Hi @usualoma !

Thanks. This is what I want. I've tested it locally it works well. With the JSX Renderer, we can write the streaming HTML response neatly.

I left the comments, please check them.

**Comment by usualoma** [comment]
@yusukebe Thank you for your review.
Done!

**Comment by yusukebe** [comment]
Thanks!

**Pull request merged**